estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,83 @@
+from odoo import api, models, fields
+from dateutil.relativedelta import relativedelta
+
+from odoo.exceptions import UserError
+from odoo.tools.float_utils import float_compare
+
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Real Estate Property Offer"
+    _order = "price desc"
+
+    price = fields.Float()
+    status = fields.Selection(
+        string='Status',
+        copy=False,
+        selection=[
+            ('accepted', 'Accepted'),
+            ('refused', 'Refused'),
+        ],
+    )
+    validity = fields.Integer(string="Validity (days)", default=7)
+    date_deadline = fields.Date(string="Deadline", compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+    partner_id = fields.Many2one("res.partner", required=True)
+    property_id = fields.Many2one("estate.property", required=True)
+    property_type_id = fields.Many2one(related="property_id.property_type_id", string="Property Type", store=True)
+
+    _sql_constraints = [
+        ('check_price', 'CHECK(price > 0)', 'The offer price must be strictly positive.')
+    ]
+
+    @api.model_create_multi
+    def create(self, vals):
+        for val in vals:
+            property_id = val.get('property_id')
+            price = val.get('price')
```

Comment:
by default, `get` will return `None` if no 2nd argument is specified but you assume they exist later in the code.
```suggestion
            property_id = val.get('property_id')
            price = val.get('price')
```

Reply:
ici, dans ce cas, qu'est ce que je devrais  faire pour palier à ce problème 
est ce que je dois mettre une valeur par défaut ou afficher une erreur quand le get return None

Reply:
Je pense qu'une valeur par défaut fait sens pour `price`
Pour `property_id` je pense qu'ici il est `require=True` (et n'a pas de `default`) donc je pense tu devrais déjà avoir une erreur (dans le `super().create` probablement) si elle n'est pas précisée. Mais j'aurais quand même tendance à skip un partie de la logique avec une condition par exemple